clang-tools-extra/test/clang-tidy/checkers/bugprone/capturing-this-by-field.cpp

```diff
@@ -0,0 +1,144 @@
+// RUN: %check_clang_tidy -std=c++11-or-later %s bugprone-capturing-this-by-field %t -- -config="{CheckOptions: {bugprone-capturing-this-by-field.FunctionWrapperTypes: '::std::function;::Fn'}}" --
+
```

Comment:
Please add test with a lambda that doen's capture `this`. In order to make sure that no warning will be fired